enterprise/tools/upload_local_snapshot/upload_local_snapshot.go

```diff
@@ -0,0 +1,274 @@
+// so it can be more easily inspected and run with vmstart, for example.
+
+var (
+	cacheTarget  = flag.String("cache_target", "grpcs://remote.buildbuddy.dev", "The remote cache target to upload the snapshot to.")
+	filecacheDir = flag.String("file_cache_dir", "/tmp/buildbuddy/filecache", "The local filecache dir on the executor")
+	apiKey       = flag.String("api_key", "", "The API key to use to interact with the remote cache.")
+	// Note: this key can be copied and pasted from logs:
+	//
+	//  "INFO: Found snapshot for key {...}" // copy this JSON
+	//
+	// At the shell, paste it in single quotes: -remote_snapshot_key='<paste>'
+	remoteSnapshotKeyJSON = flag.String("remote_snapshot_key", "", "JSON struct containing a local snapshot key that should be uploaded to the remote cache.")
+)
+
+type RemoteSnapshotKey struct {
+	GroupID      string `json:"group_id"`
+	InstanceName string `json:"instance_name"`
+	KeyDigest    string `json:"key_digest"`
+	Key          any    `json:"key"`
+}
+
+func main() {
+	flag.Parse()
+
+	if *remoteSnapshotKeyJSON == "" {
+		log.Fatalf("You must pass in a snapshot key")
+	}
+	inputKey, key, err := parseSnapshotKeyJSON(*remoteSnapshotKeyJSON)
+	if err != nil {
+		log.Fatalf("Failed to parse snapshot key: %s", err)
+	}
+	snapshotGroupID := inputKey.GroupID
+	snapshotInstanceName := inputKey.InstanceName
+
+	ctx := context.Background()
+	if *apiKey != "" {
+		ctx = metadata.AppendToOutgoingContext(ctx, "x-buildbuddy-api-key", *apiKey)
+	}
+	env := getToolEnv()
+	loader, err := snaploader.New(env)
+	if err != nil {
+		log.Fatalf("Failed to init snaploader: %s", err)
+	}
+
+	allSnapshotDigests := getAllDigestsFromSnapshotManifest(ctx, env, snapshotInstanceName, snapshotGroupID, key)
+	missingDigestsRemoteCache := getMissingDigestsRemoteCache(ctx, env, snapshotInstanceName, allSnapshotDigests)
+	uploadDigestsRemoteCache(ctx, env, snapshotInstanceName, missingDigestsRemoteCache)
+	uploadManifestRemoteCache(ctx, env, key, snapshotGroupID, snapshotInstanceName)
+
+	// Sanity check that snapshot exists in remote cache
+	flagutil.SetValueForFlagName("executor.enable_remote_snapshot_sharing", true, nil, false)
+	_, err = loader.GetSnapshot(ctx, &fcpb.SnapshotKeySet{BranchKey: key}, true)
+	if err != nil {
+		log.Fatalf("Snapshot upload failed. Snapshot is not in the remote cache after it should've been uploaded.")
+	}
+
+	log.Infof("Snapshot successfully uploaded!")
+}
+
+func hashStrings(strs ...string) string {
+	out := ""
+	for _, s := range strs {
+		out += hash.String(s)
+	}
+	return hash.String(out)
+}
+
+func getToolEnv() *real_environment.RealEnv {
+	healthChecker := healthcheck.NewHealthChecker("upload-local-snapshot")
+	re := real_environment.NewRealEnv(healthChecker)
+
+	// Set a large file cache size so this tool doesn't evict anything from
+	// the executor's filecache
+	// This is set to match the prod file cache size
+	fcSize := int64(1_250_000_000_000)
```

Comment:
let's set it much larger to avoid evicting even if we change the prod cache size

```suggestion
	fcSize := int64(1e18)
```

Reply:
Done